core/module.cc

```diff
@@ -27,7 +110,18 @@ Module *ModuleBuilder::CreateModule(const std::string &name,
 }
 
 bool ModuleBuilder::AddModule(Module *m) {
-  return all_modules_.insert({m->name(), m}).second;
+  if (m->is_task_) {
+    if (!tasks_.insert(m->name()).second) {
+      return false;
+    }
+  }
+  bool module_added = all_modules_.insert({m->name(), m}).second;
+  bool node_added =
```

Comment:
Need to check that module was added before adding node, e.g.:
`
if (!module_added) {
  return false;
}
`